traffic_ops/traffic_ops_golang/api/generic_crud.go

```diff
@@ -156,6 +166,37 @@ func GenericUpdate(val GenericUpdater) (error, error, int) {
 	return nil, nil, http.StatusOK
 }
 
+// GenericOptionsDelete does a Delete (DELETE) for the given GenericOptionsDeleter object and type. Unlike
+// GenericDelete, there is no requirement that a specific key is used as the parameter.
+// GenericOptionsDeleter.DeleteKeyOptions() specifies which keys can be used.
+func GenericOptionsDelete(val GenericOptionsDeleter) (error, error, int) {
+	where, _, _, queryValues, errs := dbhelpers.BuildWhereAndOrderByAndPagination(val.APIInfo().Params, val.DeleteKeyOptions())
+	if len(errs) > 0 {
+		return util.JoinErrs(errs), nil, http.StatusBadRequest
+	}
+
+	query := val.DeleteQueryBase() + where
+	tx := val.APIInfo().Tx
+	result, err := tx.NamedExec(query, queryValues)
+	if err != nil {
+		return nil, errors.New("deleting " + val.GetType() + ": " + err.Error()), http.StatusInternalServerError
+	}
+
+	if rowsAffected, err := result.RowsAffected(); err != nil {
+		return nil, errors.New("deleting " + val.GetType() + ": getting rows affected: " + err.Error()), http.StatusInternalServerError
+	} else if rowsAffected < 1 {
+		return errors.New("no " + val.GetType() + " with that key found"), nil, http.StatusNotFound
+	} else if rowsAffected > 1 {
+		err := tx.Rollback()
```

Comment:
this should be handled by the HandleError -> https://github.com/apache/trafficcontrol/blob/master/traffic_ops/traffic_ops_golang/api/api.go#L131

Reply:
Right you are, removed in 2ada94e8f2.